commons-packet/commons-packet-manager/src/main/java/io/mosip/commons/packet/impl/PacketWriterImpl.java

```diff
@@ -143,7 +143,11 @@ public void addMetaInfo(String id, String key, String value) {
                 PacketInfo packetInfo = new PacketInfo();
                 packetInfo.setProviderName(this.getClass().getSimpleName());
                 packetInfo.setSchemaVersion(new Double(version).toString());
-                packetInfo.setId(id);
+                
+                packetInfo.setId(offlineMode ? (iteration == 0 ? String.format("%s-%s", id, process)
```

Comment:
once u change to wrapper class check if iteration is null

Reply:
Modified to Wrapper class and added null check